protocol/x/perpetuals/lib/lib.go

```diff
@@ -152,5 +158,24 @@ func GetMarginRequirementsInQuoteQuantums(
 		bigQuoteQuantums,
 		openInterestQuoteQuantums, // pass in current OI to get scaled IMR.
 	)
+
+	// Apply leverage scaling if configured
+	if leverage > 0 {
+		// Calculate max leverage: 1,000,000 / InitialMarginPpm
+		if liquidityTier.InitialMarginPpm == 0 {
+			panic("InitialMarginPpm cannot be zero for leverage calculation")
+		}
+		maxLeverage := lib.OneMillion / liquidityTier.InitialMarginPpm
+
+		// Scale IMR: baseIMR * (maxLeverage / userLeverage)
+		if leverage <= maxLeverage {
+			leverageRatio := maxLeverage / leverage
```

Comment:
Rounding: 20x max leverage and 11x custom leverage would result in increase of net IMF? Another problem that'll go away after we switch to `custom_imf_ppm`